test/e2e/petset.go

```diff
@@ -216,6 +216,10 @@ var _ = framework.KubeDescribe("PetSet [Slow] [Feature:PetSet]", func() {
 			By("Creating foo:bar in member with index 0")
 			pet.write(0, map[string]string{"foo": "bar"})
 
+			// Wait for the write to complete to all slaves with a timeout of 100s (specified in ms).
+			By("waiting for all slaves to sync")
+			pet.waitForReplication(0, ps.Spec.Replicas - 1, 100000)
```

Comment:
No i mean polling on kubectl exec is good for reasons other than replication lag, and would deal with the same issue. Existing pets should be using a common interface in this e2e (`petTester`), im asking that you change the read methods on that interface.